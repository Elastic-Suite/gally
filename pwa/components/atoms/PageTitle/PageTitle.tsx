import { Typography, TypographyProps } from '@mui/material'
import { styled } from '@mui/material/styles'
import { ReactNode } from 'react'

const Root = styled('div')(() => ({
  display: 'flex',
}))

const CustomTypography = styled(Typography)(({ theme }) => ({
  position: 'relative',
  marginBottom: '48px',
  flex: '1',
  color: theme.palette.colors.neutral['900'],

  '&:after': {
    content: '""',
    display: 'block',
    position: 'absolute',
    width: '56px',
    height: '4px',
    left: '0',
    bottom: '-12px',
    background: theme.palette.primary.main,
    borderRadius: '2px',
  },
}))

interface IProps extends TypographyProps {
  children?: ReactNode
  title: string
}

function PageTitle(props: IProps): JSX.Element {
  const { children, title, ...typographyProps } = props
  return (
    <Root>
      {title ? (
        <CustomTypography {...typographyProps}>{title}</CustomTypography>
      ) : null}
      {Boolean(children) && <div>{children}</div>}
    </Root>
  )
}

PageTitle.defaultProps = {
  variant: 'h1',
}

export default PageTitle
