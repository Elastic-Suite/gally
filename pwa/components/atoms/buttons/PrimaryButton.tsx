import Button from '@mui/material/Button'
import CommonButtonStyle from "~/components/atoms/buttons/CommonButtonStyle";

const PrimaryButton = (props) => {
  return (
    <Button {...props}
      sx={CommonButtonStyle}
      variant="contained"
    >
      {props.children}
    </Button>
  )
}

export default PrimaryButton
