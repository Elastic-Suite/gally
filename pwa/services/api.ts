import { apiUrl } from '~/constants'

export function fetchApi<T>(
  language: string,
  url: RequestInfo | URL,
  options: RequestInit = {}
): Promise<T> {
  if (process.env.NEXT_PUBLIC_LOCAL) {
    if (!String(url).startsWith('/')) {
      url = `/${url}`
    }
    if (!String(url).endsWith('.json') && !String(url).endsWith('.jsonld')) {
      url = `${url}.json`
    }
    url = `/mocks${url}`
  }
  url = `${apiUrl}${url}`
  return fetch(url, {
    ...options,
    headers: {
      ...options.headers,
      'Accept-Language': language,
    },
  }).then((response) => response.json())
}
