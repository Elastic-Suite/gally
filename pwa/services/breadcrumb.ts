import { IMenuChild } from '~/store'

export function getSlugArray(data: string[] | string): string[] {
  const newBreadCrumbData = [data[0]]
  for (let index = 0; index < data.length - 1; index++) {
    newBreadCrumbData.push(`${newBreadCrumbData[index]}_${data[index + 1]}`)
  }
  return newBreadCrumbData
}

export function findBreadcrumbLabel(
  findIndex: number,
  slug: string[],
  menu: IMenuChild[],
  deepIndex = 0
): string {
  if (typeof menu === 'object') {
    for (const menuChild in menu) {
      if (menu[menuChild].code === slug[deepIndex]) {
        if (findIndex === deepIndex) {
          return menu[menuChild].label
        }
        return findBreadcrumbLabel(
          findIndex,
          slug,
          menu[menuChild].children,
          deepIndex + 1
        )
      }
    }
  }
  return null
}
