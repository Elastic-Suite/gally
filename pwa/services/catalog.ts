import { ICatalog } from '~/types'

export function findDefaultCatalog(catalogsData: ICatalog[]): ICatalog | null {
  const defaultCatalog = catalogsData
    ? catalogsData.filter((catalog) =>
        catalog.localizedCatalogs.find((localizedCtl) => localizedCtl.isDefault)
      )[0]
    : null

  if (defaultCatalog) {
    defaultCatalog.localizedCatalogs = defaultCatalog.localizedCatalogs.filter(
      (localizedCtl) => localizedCtl.isDefault
    )
  }
  return defaultCatalog ? defaultCatalog : null
}
