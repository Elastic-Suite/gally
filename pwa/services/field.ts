import { IElasticSuiteProperty, IField } from '~/types'

function updateProperties(
  properties: IElasticSuiteProperty,
  key: string,
  value: number | boolean
): IElasticSuiteProperty {
  if (key === 'visible') {
    return {
      ...properties,
      visible: value as boolean,
    }
  } else if (key === 'editable') {
    return {
      ...properties,
      editable: value as boolean,
    }
  } else if (key === 'position') {
    return {
      ...properties,
      position: value as number,
    }
  }
  return properties
}

export function updatePropertiesAccordingToPath(
  field: IField,
  path: string
): IField {
  if (path.includes('admin/settings')) {
    path = 'settings_attribute'
  } else {
    path = path.replaceAll('/', '_').replace('_admin_', '')
  }
  if (field.elasticsuite?.context) {
    const [_, newPropertiesvalues] = Object.entries(field.elasticsuite?.context)
      .filter(([key, _]) => key === path)
      .flat()
    if (newPropertiesvalues) {
      let newProperties = field.elasticsuite
      // eslint-disable-next-line no-return-assign
      Object.entries(newPropertiesvalues).forEach(
        ([property, propertyValue]) =>
          (newProperties = updateProperties(
            newProperties,
            property,
            propertyValue
          ))
      )
      return {
        ...field,
        elasticsuite: newProperties,
      }
    }
  }
  return field
}
