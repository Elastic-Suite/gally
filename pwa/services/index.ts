export * from './api'
export * from './authProvider'
export * from './format'
export * from './hydra'
export * from './jsonld'
export * from './local'
export * from './table'
export * from './tests'
