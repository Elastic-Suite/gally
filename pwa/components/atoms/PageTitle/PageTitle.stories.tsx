import { ComponentMeta, ComponentStory } from '@storybook/react'

import PrimaryButton from '../buttons/PrimaryButton'

import PageTitle from './PageTitle'

export default {
  title: 'Atoms/PageTitle',
  component: PageTitle,
} as ComponentMeta<typeof PageTitle>

const Template: ComponentStory<typeof PageTitle> = (args) => (
  <PageTitle {...args} />
)

export const Default = Template.bind({})
Default.args = {
  title: 'Page title',
}

export const WithButtons = Template.bind({})
Default.args = {
  children: <PrimaryButton>Import (xlsx)</PrimaryButton>,
  title: 'Page title',
}
