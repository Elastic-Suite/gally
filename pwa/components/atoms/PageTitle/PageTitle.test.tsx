import PageTitle from './PageTitle'
import { renderWithProviders } from '~/services'
import PrimaryButton from '../buttons/PrimaryButton'

describe('PageTitle match snapshot', () => {
  it('PageTitle simple', () => {
    const { container } = renderWithProviders(<PageTitle title="Hello World" />)
    expect(container).toMatchSnapshot()
  })

  it('PageTitleH2', () => {
    const { container } = renderWithProviders(
      <PageTitle title="Hello World" variant="h2" />
    )
    expect(container).toMatchSnapshot()
  })

  it('PageTitleH3WithChildren', () => {
    const { container } = renderWithProviders(
      <PageTitle title="Hello World" variant="h3">
        <PrimaryButton>Import (xlsx)</PrimaryButton>
      </PageTitle>
    )
    expect(container).toMatchSnapshot()
  })
})
