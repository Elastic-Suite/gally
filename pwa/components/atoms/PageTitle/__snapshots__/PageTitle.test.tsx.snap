// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitle match snapshot PageTitle simple 1`] = `
<div>
  <div
    class="css-k008qs"
  >
    <h1
      class="MuiTypography-root MuiTypography-h1 css-5ukfbz-MuiTypography-root"
    >
      Hello World
    </h1>
  </div>
</div>
`;

exports[`PageTitle match snapshot PageTitleH2 1`] = `
<div>
  <div
    class="css-k008qs"
  >
    <h2
      class="MuiTypography-root MuiTypography-h2 css-179nbwd-MuiTypography-root"
    >
      Hello World
    </h2>
  </div>
</div>
`;

exports[`PageTitle match snapshot PageTitleH3WithChildren 1`] = `
<div>
  <div
    class="css-k008qs"
  >
    <h3
      class="MuiTypography-root MuiTypography-h3 css-1v3h5o1-MuiTypography-root"
    >
      Hello World
    </h3>
    <div>
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-13u4ebv-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Import (xlsx)
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
